app/src/pages/inside/stepPage/modals/linkIssueModal/linkIssueModal.jsx

```diff
@@ -204,55 +245,159 @@ export class LinkIssueModal extends Component {
+      let requests;
+      if (this.isBulkOperation) {
+        testItems.map((elem) => {
+          return testItemLogRequest.push(
+            fetch(URLS.logItemStackTrace(activeProject, elem.path, ERROR_LOGS_SIZE)),
+          );
+        });
+        requests = testItemLogRequest;
+      } else {
+        testItemLogRequest = fetch(
+          URLS.logItemStackTrace(activeProject, testItems.path, ERROR_LOGS_SIZE),
+        );
+        requests = [testItemLogRequest];
+      }
+
+      Promise.all(requests)
+        .then((responses) => {
+          const [testItemRes] = responses;
+          const testItemLogs = this.isBulkOperation
+            ? responses.map((item) => item.content)
+            : testItemRes.content;
+          const items = [];
+          this.isBulkOperation
+            ? testItems.map((elem, i) => {
+                return items.push({ ...elem, logs: testItemLogs[i] });
+              })
+            : items.push({ ...testItems, logs: testItemLogs });
+          this.setState({
+            modalState: {
+              ...this.state.modalState,
+              testItems: items,
+            },
+            loading: false,
+          });
+        })
+        .catch(() => {
+          this.setState({
+            testItems: [],
+            selectedItems: [],
+            loading: false,
+          });
+          this.props.showNotification({
+            message: intl.formatMessage(messages.linkIssueFailed),
+            type: NOTIFICATION_TYPES.ERROR,
+          });
+        });
+    };
+    fetchLogs();
+  }
+
+  renderIssueFormHeaderElements = () => {
+    const {
+      intl: { formatMessage },
+      invalid,
+    } = this.props;
+    return (
+      <>
+        <GhostButton
+          onClick={() => this.props.hideModalAction()}
+          disabled={false}
+          transparentBorder
+          transparentBackground
+          appearance="topaz"
+        >
+          {formatMessage(messages.cancel)}
+        </GhostButton>
+        <GhostButton onClick={this.onLink} disabled={invalid} color="''" appearance="topaz">
+          {formatMessage(messages.linkIssue)}
+        </GhostButton>
+      </>
+    );
+  };
+  renderTitle = (collapsedRightSection) => {
+    const {
+      data: { items },
+      intl: { formatMessage },
+    } = this.props;
+    return collapsedRightSection
+      ? formatMessage(messages.linkIssueForTheTest, {
+          launchNumber: items.launchNumber && `#${items.launchNumber}`,
+        })
+      : formatMessage(messages.linkIssue);
+  };
+
+  setModalState = (newModalState) => {
+    this.setState({
+      modalState: {
+        ...this.state.modalState,
+        ...newModalState,
+      },
+    });
+  };
+
+  renderRightSection = (collapsedRightSection) => {
+    const {
+      modalState: { testItems, selectedItems },
+    } = this.state;
+    return (
+      <div className={cx('items-list')}>
+        <ItemsList
+          setModalState={this.setModalState}
```

Comment:
It would be better to rename `setModalState` to smth relative business function, f.e. `setItems`

Reply:
This component is used in MakeDecision modal window and its name corresponds.

Reply:
Actually, not corresponds :) You provide details of realization to ItemList. ItemList should know nothing about modal. It should just pass changed values via callback - nether mind where they will be used